### PR TITLE
Refactor documentation

### LINKED ISSUE #126: Restructuring of the fpm docs
I had a look at the fpm docs recently and I believe we need to rethink how the fpm documentation is layed out. There are a few sections that I think need to be worked on.
This is somewhat urgent due to an upcoming blog post from the openmp folks.

### Has sections for both `fpm.toml` & `config.toml`

Meant to [define](https://fpm.fortran-lang.org/spec/manifest.html#global-config-file:~:text=The%20fpm.toml%20file%20for%20each%20project%20is%20called%20its%20manifest) the fields of the `fpm.toml`, now also contains information about the [global configuration file](https://fpm.fortran-lang.org/spec/manifest.html#global-config-file). This should be a separate page _Global Configuration File_ (or something along these lines).

### _Registry settings_ section

The registry settings, API expectations and directory structures similarly are not for the `fpm.toml` but the `config.toml` and should not be part of the Manifest page. They should live under either a _Registry_ page or _Global Configuration File_.

### _Specifying dependencies_ section

This section is completely mangled and out of order. The information that it contains are not representative of what the user is expected of doing if they want to add dependencies in their projects. A lot of the core information has instead moved under the _Registry settings_ section, which I believe was a mistake. The same applies to the [_Development dependencies_](https://fpm.fortran-lang.org/spec/manifest.html#development-dependencies) section which is also under _Registry settings_.

### Outdated _Adding dependencies_ section

The section [_Adding dependencies_](https://fpm.fortran-lang.org/tutorial/dependencies.html) specifically the stdlib subsection is outdated and does not work anymore.

## 2. _Package upload_ page

I think since the [page](https://fpm.fortran-lang.org/spec/publish.html) is less about fpm and more about the registry functionality through fpm, this page should live somewhere under a _Registry_ page and not alongside the fpm manifest, API and metapackages. The same case can be made for [_Module name requirements_](https://fpm.fortran-lang.org/spec/naming.html) which is geared towards defining rules for valid names for packages in the registry.

## 3. _Registry_

Since the STF work on a registry was completed and documented in the fpm docs, now the meaning of the registry in the docs carries a dual meaning, the new not yet fully online registry and the old abandoned registry. fpm docs contains references to both. I am of the opinion that the old [_Registry_](https://fpm.fortran-lang.org/registry/index.html) page should be removed to minimise confusion from the readers, or better yet replaced with a section that reflects the current registry functionality.

## 4. Versioning the docs

Since we are introducing breaking changes e.g. with the metapackages I think it would be a good idea to start thinking about versioning the fpm docs. The obvious solution for open-source projects is [readthedocs](https://about.readthedocs.com/pricing/#/community) alternatively we could do something like SciPy's API [docs](https://docs.scipy.org/doc/scipy/reference/index.html#scipy-api) although it is substantially more elaborate to get it to work.

---

## Edit Suggestions

1. Move all of registry stuff into another page under _References_ or replace existing [_Registry_](https://fpm.fortran-lang.org/registry/index.html) page (at top level) so that _Manifest reference_ contains only references to `fpm.toml`. I personally prefer the latter approach (replace existing _Registry_ section).
2. Move Global config options from _Manifest_ into a separate page under _Reference_ with cross-reference links to both _Registry_ and _Manifest>Specifying dependencies_
3. Move section for dependencies (currently inside [Manifest>Registry](https://fpm.fortran-lang.org/spec/manifest.html#dependencies-from-version-control-systems)) inside and at the top of [_Specifying dependencies_](https://fpm.fortran-lang.org/spec/manifest.html#specifying-dependencies) with cross-references to the new _Registry_ page
4. [_Package upload_](https://fpm.fortran-lang.org/spec/publish.html) and [_Module name requirements_](https://fpm.fortran-lang.org/spec/naming.html) move inside the _Registry_ page
5. _Tutorial_ updated to reflect metapackages

---

@awvwgk and @LKedward what do you think about the proposed changes, do they seem reasonable?
I will get started on a PR to see how the changes look.

### PR DESCRIPTION
**Pull request by gnikit**
Fixes #126 _(see issue for changes)_
## Changes

_See #126_

## Minor Changes

- Define `myst_heading_anchors = 5` to generate implicit anchors (previously not generated in Manifest), resolves warnings.
- Fixes `{code-block} sh` definitions, resolves warnings.

## TODO

- [x] Section cross-references

### CODE REVIEW

**Comment by gnikit** [comment]
#preview

File: pages/tutorial/dependencies.md [hunk 1]
```diff
@@ -130,7 +130,7 @@ Our final main program looks like
 Again we run a quick check using fpm
 
 ```{code-block} text
-:emphasize-lines: 2, 14
+:emphasize-lines: 2, 13
```
Comment (awvwgk) [inline pages/tutorial/dependencies.md:133]: Here the stdlib usage needs to be changed as well

Comment (gnikit) [reply pages/tutorial/dependencies.md:133]: Good catch

**Comment by gnikit** [comment]
I pinged you all because of the rewrite of the _Registry_ section which completely removes the old registry and places the options for the new one.

**Comment by gnikit** [comment]
ping @fortran-lang/admins, could you guys have a quick look and let me know if we should include/change something in the new docs? The folks over at OpenMP are finishing their blog post about fpm and it would be good if our docs were updated before that.

**Review by milancurcic — APPROVED** [review]
I haven't tested locally but the description of changes look good to me. Thank you for this work.

**Pull request merged**